ckan/ckanext/ckanext-ytp_main/ckanext/ytp/views.py

```diff
@@ -118,6 +124,184 @@ def recent_datasets_feed():
         navigation_urls=navigation_urls)
 
 
+
+
+
+
+def _tag_string_to_list(tag_string):
+    """This is used to change tags from a sting to a list of dicts.
+    """
+    out = []
+    for tag in tag_string.split(u','):
+        tag = tag.strip()
+        if tag:
+            out.append({u'name': tag, u'state': u'active'})
+    return out
+
+
+
+
+# Based on search() from ckan 2.9
+def general_search():
+
+    extra_vars = {}
+
+    context = {
+        u'model': model,
+        u'user': g.user,
+        u'auth_user_obj': g.userobj
+    }
+
+    # unicode format (decoded from utf8)
+    extra_vars[u'q'] = q = request.args.get(u'q', u'')
+    extra_vars['query_error'] = False
+
+    # Get the page number from parameters if it's provided (maybe this can be removed?)
+    page = int(request.params.get('page', 1))
+    dataset_limit = int(config.get(u'ckan.datasets_per_page', 20))
+    organization_limit = int(config.get(u'ckan.organizations_per_page', 5))
+    sort_by = request.args.get(u'sort', "score desc, metadata_created desc")
+
+    chosen_filter = "all"
+    # Usually not passed in this way, instead submitted in the form 
+    # (we also treat organizations as a dataset type in this scenario)
+    dataset_type = request.args.get(u'dataset_type', 'all')
+
+    # Post request contains filters and page numbers that the user has selected
+    if request.method == 'POST':
+        # Get the page number from the request
+        page = int(request.form.get('page', 1))
+        sort_by = request.form.get('sort', "score desc, metadata_created desc")
+        chosen_filter = request.form.get('filter', 'all')
+        dataset_type = request.form.get('filter', 'all')
+
+    allowed_sorting = [
+        'score desc, metadata_created desc',
+        'title_string asc',
+        'title_string desc',
+        'metadata_modified desc',
+        'metadata_created asc',
+        'metadata_created desc',
+        'views_recent desc'
+    ]
+
+    sort_string = sort_by if sort_by in allowed_sorting else "score desc, metadata_created desc"
+
+    # Organizations require a bit different search term and treatment if they are the only thing searched
+    if dataset_type == 'organization':
+        # if user is searching only for organizations, we want to display more organizations than 5
+        organization_limit = 20
+        dataset_limit = 0
```

Comment:
Can't set this to zero as it is used as a divider later (and it causes an error)

Reply:
Resolved in [92a479a](https://github.com/vrk-kpa/opendata/pull/1774/commits/92a479a6671ee9d1f0071c475d1135bf7a17fbd2)